File: app/src/main/java/azuwaredison89/gmail/dinacom/ui/activity/pesanperawat/DaftarNamaPerawatActivity.java
package azuwaredison89.gmail.dinacom.ui.activity.pesanperawat;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ListAdapter;
import android.widget.ListView;
import android.widget.SimpleAdapter;

import java.util.ArrayList;
import java.util.HashMap;

import azuwaredison89.gmail.dinacom.R;

public class DaftarNamaPerawatActivity extends AppCompatActivity {

    protected ListView lv_perawat;
    protected ListAdapter adapter1;
    SimpleAdapter Adapter_perawat;
    HashMap<String, String> map_perawat;
    ArrayList<HashMap<String, String>> list_perawat;
    String[] perawat1;
    String[] perawat2;
    String[] perawat3;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_daftar_nama_perawat);
        lv_perawat = (ListView) findViewById(R.id.listView_perawat);

        perawat1 = new String[] {"Perawat Sinta", "Perawat Susanti", "Perawat Hetty", "Perawat Cenni", "Perawat Siska"};
        perawat2= new String[] {"Jam Kerja : 08.00 - 12.00", "Jam Kerja : 12.00 - 17.00",
                "Jam Kerja : 19.00 - 00.00",
                "Jam Kerja : 08.00 - 16.00", "Jam Kerja : 17.00 - 00.00"};
        perawat3 = new String[] {Integer.toString(R.drawable.perawat3),
                Integer.toString(R.drawable.perawat3),
                Integer.toString(R.drawable.perawat3),
                Integer.toString(R.drawable.perawat3),
                Integer.toString(R.drawable.perawat3)};

        list_perawat = new ArrayList<HashMap<String,String>>();

        for (int i = 0; i < perawat1.length; i++){
            map_perawat = new HashMap<String, String>();
            map_perawat.put("list", perawat1[i]);
            map_perawat.put("latin", perawat2[i]);
            map_perawat.put("gbr", perawat3[i]);
            list_perawat.add(map_perawat);
        }

        Adapter_perawat = new SimpleAdapter(this, list_perawat, R.layout.list_nama_perawat,
                new String[] {"list", "latin", "gbr"}, new int[] {R.id.judul_perawat, R.id.ltn_perawat, R.id.iv_perawat});

        lv_perawat.setAdapter(Adapter_perawat);

        lv_perawat.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            public void onItemClick(AdapterView<?> parent, View view,
                                    int position, long id) {
                String nama_negara=lv_perawat.getItemAtPosition(position).toString();
                Intent intent= new Intent(getApplicationContext(),
                        DataPerawatActivity.class);
                startActivity(intent);
                finish();
            }
        });
    }
}

File: app/src/main/java/azuwaredison89/gmail/dinacom/ui/activity/pesanperawat/DaftarNamaPerawatActivity.kt
package azuwaredison89.gmail.dinacom.ui.activity.pesanperawat

import android.content.Intent
import android.os.Bundle
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import azuwaredison89.gmail.dinacom.R
import azuwaredison89.gmail.dinacom.model.Perawat
import azuwaredison89.gmail.dinacom.ui.activity.pesanperawat.DataPerawatActivity.Companion.EXTRA_PERAWAT
import azuwaredison89.gmail.dinacom.ui.adapter.ListPerawatAdapter
import com.google.firebase.database.DataSnapshot
import com.google.firebase.database.DatabaseError
import com.google.firebase.database.FirebaseDatabase
import com.google.firebase.database.ValueEventListener
import kotlinx.android.synthetic.main.activity_daftar_nama_perawat.*


class DaftarNamaPerawatActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_daftar_nama_perawat)
        getDataPerawat()
    }

    private fun getDataPerawat() {
        val listPerawat = ArrayList<Perawat>()
        val db = FirebaseDatabase.getInstance()
        val rf = db.getReference("Perawat")
        rf.addValueEventListener(object : ValueEventListener {
            override fun onCancelled(p0: DatabaseError) {
                Toast.makeText(this@DaftarNamaPerawatActivity, "Failed Load Data Perawat", Toast.LENGTH_SHORT).show()
            }

            override fun onDataChange(dataSnapshot: DataSnapshot) {
                for (snap in dataSnapshot.children) {
                    val perawat = snap.getValue(Perawat::class.java)!!
                    listPerawat.add(perawat)
                    recyclerViewPerawat.apply {
                        layoutManager = LinearLayoutManager(this@DaftarNamaPerawatActivity)
                        setHasFixedSize(true)
                        adapter = ListPerawatAdapter(listPerawat).apply {
                            setOnClickCallback(object : ListPerawatAdapter.OnItemCallback{
                                override fun onItemClicked(data: Perawat) {
                                    val intent =  Intent(this@DaftarNamaPerawatActivity, DataPerawatActivity::class.java).apply {
                                        putExtra(EXTRA_PERAWAT, perawat.nama)
                                    }
                                    startActivity(intent)
                                }

                            })
                        }
                    }
                }
            }
        })
    }
}